src/python/deepgnn/graph_engine/snark/tests/convert_test.py

```diff
@@ -46,7 +60,12 @@ def triangle_graph_json(folder):
             "node_weight": 1,
             "neighbor": {"0": {}, "1": {"5": 1}},
             "uint64_feature": {},
-            "float_feature": {"0": [1], "1": [-0.03, -0.04]},
+            "float_feature": [[1], [-0.03, -0.04]]
+            if formatType == FeatureFormat.Array
+            else {"0": [1], "1": [-0.03, -0.04]},
+            "double_feature": {"0": [1], "1": [-0.03, -0.04]}
```

Comment:
What is the expected return value when we fetch features with id = 0, if we have 2 features with the same id, but different types, float and double?

Reply:
For each feature we have only 1 unique id. If there are features with same id, we will assign a non-existing id for it.

Reply:
By assigning a hidden id is just a tradeoff, in case that users make mistake by setting ids. We could suggest users to use array or TSV format to convert the data.

The test cases are covered [here](https://github.com/microsoft/DeepGNN/blob/tiantiaw/update_converter/src/python/deepgnn/graph_engine/snark/tests/convert_test.py#:~:text=param%20%3D%20%5B,%5D).

Reply:
> A new id is hidden in a conversion log? It will be hard to find it for a graph with billions of nodes...
Any other solutions considered?

By default user should care about the index and make sure it is correct, this approach is just to make sure the conversion job will not crash or hang. Converter will show warning log for users to identify the correct index assigned.

My suggestion is that user just need to provide an array instead of map, and converter do `array->map` internally, or users can use TSV format as raw graph file.

Reply:
> Where are tests for feature fetching with the new ids?
> 
> The test cases are covered [here](https://github.com/microsoft/DeepGNN/blob/tiantiaw/update_converter/src/python/deepgnn/graph_engine/snark/tests/convert_test.py#:~:text=param%20%3D%20%5B,%5D).

I don't see any feature fetching in these tests.

> My suggestion is that user just need to provide an array instead of map, and converter do array->map internally, or users can use TSV format as raw graph file.

Why are we not doing it?

Reply:
The purpose of this PR is try to simplify the user's converter job even when the raw graph is a wrong version, however, when I start doing this PR, I also noticed that maybe it is not a good idea to have a hidden id to replace the error id user provided. It is a tradeoff and the disadvantage of this PR is much bigger than the benefit. Actually, user should guarantee the correctness of the raw graph file, if they don't want to use json, they can use TSV because in TSV there is no need to generate such feature ids. Let me revert this PR, thanks.